admittance_controller/include/admittance_controller/admittance_rule_impl.hpp

```diff
@@ -185,37 +155,52 @@ controller_interface::return_type AdmittanceRule::update(
     apply_parameters_update();
   }
 
-  bool success = get_all_transforms(current_joint_state, reference_joint_state);
+  bool success = true;
+
+  // Reusable transform variable
+  Eigen::Isometry3d tf;
+
+  // --- FT sensor frame to base frame (translation + rotation) ---
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.ft_sensor.frame.id, tf);
+  admittance_state_.ref_trans_base_ft = tf;
 
-  // apply filter and update wrench_world_ vector
-  Eigen::Matrix<double, 3, 3> rot_world_sensor =
-    admittance_transforms_.world_base_.rotation() * admittance_transforms_.base_ft_.rotation();
-  Eigen::Matrix<double, 3, 3> rot_world_cog =
-    admittance_transforms_.world_base_.rotation() * admittance_transforms_.base_cog_.rotation();
-  process_wrench_measurements(measured_wrench, rot_world_sensor, rot_world_cog);
+  // --- world frame to base frame (we only need the rotation) ---
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.fixed_world_frame.frame.id, tf);
+  Eigen::Matrix3d rot_world_base = tf.rotation();
 
-  // transform wrench_world_ into base frame
+  // --- control/base frame to base frame (rotation only) ---
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.kinematics.base, tf);
```

Comment:
```suggestion
    current_joint_state.positions, parameters_.control.frame.id, tf);
```
Shouldn't it be this?

https://github.com/ros-controls/ros2_controllers/blob/945a3609978bd67ba68dc48c2f1e3465301a0f8d/admittance_controller/include/admittance_controller/admittance_rule_impl.hpp#L167-L169